<?xml version="1.0" encoding="utf-8"?>	

<launch>

  <arg name="robot_name" default="robot" />
  <arg name="simulator" value="" />
  
  <arg name="respawn_value" default="false" /> <!-- boolean: true, false -->
  <arg name="use_marker_controller" default="true" /> <!-- boolean: true, false -->
  
  <arg name="enable_laser_proximity_callback" default="false"/>  <!-- enable a more conservative use of laser proximity info -->
  <arg name="lambda_trav" default="1.0"/>  

  <node name="path_planner_manager_$(arg robot_name)" pkg="path_planner" type="path_planner_manager" respawn="$(arg respawn_value)" output="screen"> 

        <param name = "robot_frame_name" value = "base_link"/>
        <param name = "int_marker_server_name" value = "/marker_controller"/>
        <param name = "int_marker_name" value = "$(arg robot_name)"/>
         
        <param name = "lambda_trav"  value="$(arg lambda_trav)"/>  
        
        <param name = "use_marker_controller" value = "$(arg use_marker_controller)"/>
        
        <param name = "path_planning_service_name" value = "/path_planning_service"/>
        
        <param name = "enable_laser_proximity_callback" value = "$(arg enable_laser_proximity_callback)"/>
            
        <!--param name = "goal_topic_name" value = "/goal_topic"/-->
        <!--param name = "goal_abort_topic_name" value = "/goal_abort_topic_name"/-->

        <!-- 
        <remap from = "/clustered_pcl/wall" to = "/clustered_pcl/wall"/>
        <remap from = "/trav/traversability" to = "/trav/traversability"/> 
        
        <remap from = "/goal_topic" to = "/goal_topic"/>
        <remap from = "/goal_abort_topic" to = "/goal_abort_topic"/>
        <remap from = "/trajectory_control_abort_topic" to = "/trajectory_control_abort_topic"/>
        
        <remap from = "/robot_path" to = "/robot_path"/>
        <remap from = "/robot_local_path" to = "/robot_local_path"/>
        
        <remap from = "/robot_path_draw" to = "/robot_path_draw"/>
        <remap from = "/robot_local_path_draw" to = "/robot_local_path_draw"/>

        <remap from = "/path_planner/visited_nodes" to = "/path_planner/visited_nodes"/>
        <remap from = "/path_planner/localPath" to = "/path_planner/localPath"/>
        
        <remap from = "/planner/tasks/path" to = "/planner/tasks/path"/>
        <remap from = "/planner/tasks/global_path" to = "/planner/tasks/global_path"/>
        <remap from = "/planner/tasks/feedback" to = "/planner/tasks/feedback"/>
        
        <remap from = "/path_planning_status" to = "/path_planning_status"/>
        
        <remap from = "/laser_proximity_topic" to = "/laser_proximity_topic"/> 
        -->
    
  </node> 
  
</launch>
